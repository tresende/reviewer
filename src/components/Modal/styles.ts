import styled, { css, createGlobalStyle } from 'styled-components'
import media from 'styled-media-query'

export const GlobalStyle = createGlobalStyle`
  body {
    overflow:hidden
  }
`

const WrapperModifiers = {
  close: () => css`
    pointer-events: none;
  `
}

export const Wrapper = styled.div<{ open: boolean }>`
  ${({ theme, open }) => css`
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    margin-right: ${theme.spacings.small};
    display: flex;
    background: rgba(0, 0, 0, 0.45);
    z-index: ${theme.layers.overlay};
    justify-content: center;
    align-items: center;
    ${!open && WrapperModifiers.close()};
    ${media.lessThan('small')`
      align-items: flex-end;
    `}
  `}
`

export const Modal = styled.div`
  ${({ theme }) => css`
    max-width: 600px;
    min-height: 400px;
    pointer-events: auto;
    border-radius: ${theme.border.mediumRadius};
    background-color: white;
    ${media.lessThan('small')`
      min-height: 80%;
    `}
  `}
`
