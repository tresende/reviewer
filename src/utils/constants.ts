const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MIN_TEXT_LENGTH = 5
const MAX_RATING = 5
const DEFAULT_SCORE = 3
const ANIMATION_DURATION = 1000

export { SCORE_LABEL, MAX_RATING, ANIMATION_DURATION, MIN_TEXT_LENGTH, DEFAULT_SCORE }
