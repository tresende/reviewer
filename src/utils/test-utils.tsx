import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
