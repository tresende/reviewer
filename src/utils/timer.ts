export function timeout(ms?: number | undefined) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}
