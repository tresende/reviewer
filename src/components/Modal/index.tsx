import { useEffect } from 'react'

import * as S from './styles'

export type ModalProps = {
  open?: boolean
  children?: React.ReactNode
  onClose: () => void
}

const Modal = ({ onClose, children, open }: ModalProps) => {
  useEffect(() => {
    const handleKeyUp = ({ key }: KeyboardEvent) => {
      key === 'Escape' && onClose()
    }

    window.addEventListener('keyup', handleKeyUp)
    return () => window.removeEventListener('keyup', handleKeyUp)
  }, [])
  if (open === undefined) return null

  const modalClasses = `animate__animated ${!open ? 'animate__bounceOutUp' : 'animate__bounceInUp'} `
  const backgroundClasses = `animate__animated ${!open ? 'animate__fadeOut' : 'animate__fadeIn'} `

  return (
    <S.Wrapper open={open} onClick={onClose} className={backgroundClasses} role="modal">
      <S.Modal onClick={(e) => e.stopPropagation()} className={modalClasses}>
        {children}
      </S.Modal>
      {open && <S.GlobalStyle />}
    </S.Wrapper>
  )
}

export default Modal
