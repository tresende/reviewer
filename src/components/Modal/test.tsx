import { render, screen, fireEvent } from 'utils/test-utils'
import Modal from '.'

describe('<Modal />', () => {
  it('should call onclose callback when user press escape', () => {
    const onClose = jest.fn()
    const { container } = render(<Modal onClose={onClose} />)
    fireEvent.keyUp(container, { key: 'Escape' })

    expect(onClose).toBeCalledTimes(1)
  })

  it("should't call client in form div", () => {
    const onClose = jest.fn()
    render(
      <Modal onClose={onClose} open>
        <button>Click!</button>
      </Modal>
    )
    screen.getByRole('button').click()
    expect(onClose).toBeCalledTimes(0)
  })

  it('should add fadeIn and bounceIn classes', () => {
    const onClose = jest.fn()
    const { container } = render(<Modal onClose={onClose} open />)
    expect(container.firstChild?.firstChild).toHaveClass('animate__bounceInUp')
    expect(container.firstChild).toHaveClass('animate__fadeIn')
  })

  it('should add fadeOut and bounceOut classes', () => {
    const onClose = jest.fn()
    const { container } = render(<Modal onClose={onClose} open={false} />)
    expect(container.firstChild?.firstChild).toHaveClass('animate__bounceOutUp')
    expect(container.firstChild).toHaveClass('animate__fadeOut')
  })
})
