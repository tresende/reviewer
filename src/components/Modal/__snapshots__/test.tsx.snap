// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should add fadeIn and bounceIn classes 1`] = `
.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  margin-right: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: rgba(0,0,0,0.45);
  z-index: 30;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 600px;
  min-height: 400px;
  pointer-events: auto;
  border-radius: 0.8rem;
  background-color: white;
}

@media (max-width:450px) {
  .c0 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

@media (max-width:450px) {
  .c1 {
    min-height: 80%;
  }
}

<div
  class="c0 animate__animated animate__fadeIn "
  open=""
  role="modal"
>
  <div
    class="c1 animate__animated animate__bounceInUp "
  />
</div>
`;

exports[`<Modal /> should add fadeOut and bounceOut classes 1`] = `
.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  margin-right: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: rgba(0,0,0,0.45);
  z-index: 30;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
}

.c1 {
  max-width: 600px;
  min-height: 400px;
  pointer-events: auto;
  border-radius: 0.8rem;
  background-color: white;
}

@media (max-width:450px) {
  .c0 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

@media (max-width:450px) {
  .c1 {
    min-height: 80%;
  }
}

<div
  class="c0 animate__animated animate__fadeOut "
  role="modal"
>
  <div
    class="c1 animate__animated animate__bounceOutUp "
  />
</div>
`;
